ynput/OpenPype
# Nuke: update rendered frames in latest version

## Brief description
Introduced new field to insert frame(s) to rerender only.

## Description
Rendering is expensive, sometimes it is helpful only to re-render changed frames and reuse existing.
Artists can in Publisher fill which frame(s) should be re-rendered.
![2023-01-24 17_10_45-OpenPype publisher](https://user-images.githubusercontent.com/4457962/214346380-c0c3d080-488c-4fa7-9375-04ce0acb67c2.png)


If there is already published version of currently publishing subset, all representation files are collected (currently for `render` family only) and then when Nuke is rendering (locally only for now), old published files are copied into into temporary render folder where will be rewritten only by frames explicitly set in new field.

That way review/burnin process could also reuse old files and recreate reviews/burnins.

New version is produced during this process!

## Additional info

Currently only implemented for Nuke's local render as a proof of concept. Farm rendering will be implemented afterwards.

Only other host where this workflow can work is for now AfterEffects as they have Publisher implemented.

It might be useful to create abstract extractor class and use it in host extractor as multiple inheritance to highlight that this extractor implements this functionality (kinda trait - expect some specific fields, does something)...

Contains fix for AE found during testing, didn't want to create new PR for this (expected to implement AE first, decided for Nuke instead after commits were done.)


## Testing notes:
1. publish sequence of frames in Nuke
2. change specific frame in Nuke workfile (to see change)
3. fill frame of changed frame into text field
4. publish again - check published frame, review etc.
